ocis-pkg/natsjsregistry/registry.go

```diff
@@ -150,6 +170,16 @@ func storeOptions(opts registry.Options) []store.Option {
 	natsOptions := nats.GetDefaultOptions()
 	natsOptions.Name = "nats-js-kv-registry"
 	natsOptions.User, natsOptions.Password = getAuth()
+	natsOptions.ReconnectedCB = func(_ *nats.Conn) {
+		if err := n.Init(); err != nil {
+			fmt.Println("cannot reconnect to nats")
+			os.Exit(1)
```

Comment:
:boom: OUCH

I guess I understand why you're doing that. And I don't have a better solution at hand. But I still feel the need to say: This is ugly :smile: 

Would be nice if we could find a way for a more graceful error handling here.

Reply:
Yes I 100% agree. :heart: I'll fix it as soon as we have a proper solution.